csrc/scheduler/vectorize_helper.cpp

```diff
@@ -1094,8 +1094,13 @@ std::vector<std::pair<ProjectedExtent&, IterDomain*>> getContigVectorSizesOf(
       continue;
     }
 
+    // NOTE: Are we expecting a contiguity flag here? Would a nullopt be
+    // equivalent to a False for the analysis?
+    TORCH_INTERNAL_ASSERT(
+        contiguity.at(root_i).has_value(),
+        "contiguity flat at root_i can't be null");
     // Not contiguous
-    if (!*contiguity[root_i]) {
+    if (!contiguity.at(root_i).value()) {
```

Comment:
QQ @zasdfgbnm : since the old code is somewhat ill formed, what's the expected behavior here?

I think we really just wanted `if (!contiguity.at(root_i).has_value() || !contiguity.at(root_i).value()) { break; }`, i.e. which is a strict `not contiguous` that excludes broadcast dimensions?!

Reply:
If the contiguity has no value, then the corresponding IterDomain must be broadcast, and it must have been skipped above. If not, then this is a bug. So we should `TORCH_INTERNAL_ASSERT` that contiguity has value.

Reply:
sounds like I'm already doing the right thing :sunglasses: 

I'll merge this once my local cpp tests passed.